src/NHibernate/Type/OneToOneType.cs

```diff
@@ -44,7 +44,8 @@ public override void NullSafeSet(DbCommand st, object value, int index, bool[] s
 
 		public override void NullSafeSet(DbCommand cmd, object value, int index, ISessionImplementor session)
 		{
-			//nothing to do
```

Comment:
Also, it seems that with this change the accused "hack" in hydrate is not necessary anymore.

Reply:
Hm? Which accused hack?

Reply:
This one: https://github.com/nhibernate/nhibernate-core/blob/55ffdb64d81dbc6a0b97d6c3b5a10939323ad8d6/src/NHibernate/Type/OneToOneType.cs#L98

Reply:
No. This hack should stay in place (NH1611OneToOneIdentityFixture test for it) With embedded identifiers where entity is used as ID without this hack owner entity is sent to one-to-one `ComponentType.GetPropertyValues`. And exception is occurred as one-to-one expects different type for id.

Embedded identifier is an evil that should be obsoleted.